Sources/PackageDescription4/Target.swift

```diff
@@ -61,6 +73,20 @@ public final class Target {
     /// If a value is not provided, the directory will be set to "include".
     public var publicHeadersPath: String?
 
+    public var type: TargetType
+
+    public var pkgConfig: String? {
+        didSet {
+            assert(type == .system)
+        }
+    }
+
+    public var providers: [SystemPackageProvider]? {
+        didSet {
+            assert(type == .system)
+        }
+    }
```

Comment:
It might be worth making a "payload type" (SystemLibraryTargetInfo) and changing the type to be an enum. This would get rid of the asserts

Reply:
enums are very painful w.r.t serialization but that is better since this API will have to be public.